src/node_port.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
```

Comment:
copied over from Kafka with some modifications / generalisations

Lets try to carve out some time to add this to [data platform helpers pip](https://github.com/canonical/data-platform-helpers)

Reply:
I agree that it would be a great addition to DP Helpers